packages/zipkin/index.d.ts

```diff
@@ -80,7 +80,7 @@ declare namespace zipkin {
 
       map<V>(fn: (value: T) => V): IOption<V>;
       ifPresent<V>(fn: (value: T) => V): IOption<V>;
-      flatMap<V>(fn: (value: T) => V): IOption<V>;
+      flatMap<V>(fn: (value: T) => IOption<V>): IOption<V>;
       getOrElse<V>(fnOrValue: (() => V) | V): T;
```

Comment:
Hmmm, I think getOrElse's type may be incorrect as well.  For example, None can't possibly return a T, since None represents the absence of a T.  I see two possible ways to go:

1. `getOrElse(fnOrValue: (() => T) | T): T;`  Personally I think this more rigorous type is preferable, as it makes the strong guarantee to the caller that he/she will definitely get a T: Some can return the T it already has, and None can return a T via the argument.
2. `getOrElse<V>(fnOrValue: (() => V | V): T | V;`  TS callers will be forced (and JS callers should be advised) to test the return value to know whether they got a T or a V.

The current Some and None implementations are compatible with both of those.  Let me know what you think.

Reply:
I also like the first one more, I think it is the purpose of an interface to provide the user with a common rigid signature otherwise the user may tend to implement the wrong abstraction.